reco_utils/nni/nni_utils.py

```diff
@@ -1,31 +1,32 @@
 import json
 import os
 import requests
+import subprocess
+import sys
 import time
 
-NNI_REST_ENDPOINT = 'http://localhost:8080/api/v1/nni'
-NNI_STATUS_URL = NNI_REST_ENDPOINT + '/check-status'
-NNI_TRIAL_JOBS_URL = NNI_REST_ENDPOINT + '/trial-jobs'
+
+NNI_REST_ENDPOINT = "http://localhost:8080/api/v1/nni"
+NNI_STATUS_URL = NNI_REST_ENDPOINT + "/check-status"
+NNI_TRIAL_JOBS_URL = NNI_REST_ENDPOINT + "/trial-jobs"
 WAITING_TIME = 20
-MAX_RETRIES = 5
+MAX_RETRIES = 10
 
 
-def get_experiment_status(status_url):
-    """
-    Helper method. Gets the experiment status from the REST endpoint
+def get_experiment_status(status_url=NNI_STATUS_URL):
```

Comment:
Should we also use this function to propagate any errors in the trials? https://github.com/microsoft/nni/issues/1082#event-2345623807
Maybe add a new issue about this?

Reply:
you mean `nnictl trial ls`? can you elaborate on what is in your mind about this?

Reply:
This, or by calling the REST API on NNI_TRIAL_JOBS_URL like in `get_trials()`. The trial status should be included in the response for all trials.

Reply:
added issue https://github.com/microsoft/recommenders/issues/794